classes/services/NavigationMenuService.inc.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+/**
+ * @file classes/services/NavigationMenuService.inc.php
+ *
+ * Copyright (c) 2014-2017 Simon Fraser University
+ * Copyright (c) 2000-2017 John Willinsky
+ * Distributed under the GNU GPL v2. For full terms see the file docs/COPYING.
+ *
+ * @class NavigationMenuService
+ * @ingroup services
+ *
+ * @brief Helper class that encapsulates NavigationMenu business logic
+ */
+
+namespace OJS\Services;
+
+class NavigationMenuService extends \PKP\Services\PKPNavigationMenuService {
+
+	/**
+	 * Initialize hooks for extending PKPSubmissionService
+	 */
+    public function __construct() {
+
+		\HookRegistry::register('NavigationMenus::itemTypes', array($this, 'getMenuItemTypesCallback'));
+		\HookRegistry::register('NavigationMenus::displaySettings', array($this, 'getDisplayStatusCallback'));
+	}
+
+	/**
+	 * Return all default navigationMenuItemTypes.
+	 * @param $types array of types
+	 */
+	public function getMenuItemTypesCallback($hookName, $args) {
+		$types =& $args[0];
+
+		\AppLocale::requireComponents(LOCALE_COMPONENT_APP_COMMON, LOCALE_COMPONENT_PKP_USER, LOCALE_COMPONENT_PKP_EDITOR);
+
+		$ojsTypes = array(
+			NMI_TYPE_CURRENT => array(
+				'title' => __('editor.issues.currentIssue'),
+				'description' => __('manager.navigationMenus.current.description'),
+			),
+			NMI_TYPE_ARCHIVES => array(
+				'title' => __('navigation.archives'),
+				'description' => __('manager.navigationMenus.archives.description'),
+			),
+		);
+
+		$types = array_merge($types, $ojsTypes);
+	}
+
+	/**
+	 * Callback for display menu item functionallity
+	 */
+	function getDisplayStatusCallback($hookName, $args) {
+		$navigationMenuItem =& $args[0];
+
+		$request = \Application::getRequest();
+		$dispatcher = $request->getDispatcher();
+
+		$isUserLoggedIn = \Validation::isLoggedIn();
+		$isUserLoggedInAs = \Validation::isLoggedInAs();
+		$context = $request->getContext();
+		$currentUser = $request->getUser();
+
+		$contextId = $context ? $context->getId() : CONTEXT_ID_NONE;
+
+		// Transform an item title if the title includes a {$variable}
```

Comment:
Pull the title transformation code out into a method in `PKPNavigationMenuService` so it's not duplicated here.

Reply:
OK